fecgov/fec-cms
# FIx pytest deprecation warnings, address skipped tests

## Summary
- Changes to `urls.py` to handle Pytest deprecation warnings (fec, data, legal) 
    - (https://docs.djangoproject.com/en/3.2/ref/urls/#re-path)
- Changes to `latest_updates.html` to replace  deprecated  `{% if notequal %}`
- Update requirements : `github3`, `python-dateutil`, `django_jinja` to resolve deprecation warnings for these libraries
    - for  github3: change `to_json` - to  -  `as_dict` in feedback/reaction views to fix errors in comment below
(https://github.com/fecgov/fec-cms/pull/5478#issuecomment-1310520854))
- Changes to `fec-cms/.coveragerc` to resolve failures when Pytest is run from `/fec-cms` directory
- Changes to `.pytest.ini` : 
      - to report info on Skipped or failed  tests (`--ra` flag )
       - Remove `flake8-exclude` : unrecognized option that generates a warning
- Remove unneeded, redundant JS constructor stmt that came to light when testing feedback-box 

- Resolves #5460

### Required reviewers
one-frontend, one backend  (@cnlucas please)

## Impacted areas of the application

General components of the application that this PR will affect:

Modified:  pytest.ini
Modified:  .coveragerc
Modified:  fec/data/urls.py
Modified:  fec/fec/urls.py
Modified:  fec/legal/urls.py
Modified:  requirements.txt
Modified: data/views.py
Modified:   fec/home/templates/home/latest_updates.html
Modified:   fec/static/js/pages/reaction-box.js

## How to test
This branch is pushed to dev space
- Run `pytest` on the develop branch to see the number and type of warnings 
- checkout `feature/5460-fix-pytest-deprecation-warnings`
- `pip install -r requirements.txt`
- `npm run build-js`
- run pytest from `fec-cms/` directory
- run pytest from `fec-cms/fec/` directory
- Run  `pytest --cache-clear` (This will not skip files previously passed by  Flake8) 
     - Should we consider adding `--cache-clear` to the `pytest.ini` `addopts=` flags so it it always run without  cache ?
- Test general site and admin functionality. 
    - Test feedback box and reaction box because they  use Github 3 (There  is a temp test reaction box on stage-- /data/elections/house) 
    - Test  `http://127.0.0.1:8000/updates/` to make sure querystring is correct for pagination nav btns  at bottom of page when filtering results.   Example: `http://127.0.0.1:8000/updates/?page=1&update_type=press-release&category=audit-reports`
 
- [DONE ] **Remove dev deploy task before merge and revert  h/s overview temmplates to  remove test reaction box**
